### ISSUE TITLE
Crash encountered using UXCam SDK v3.6.16

### ISSUE THREAD

**Issue by srbalan** [uxcam/uxcam-ios-sdk#14]
## Description
We are encountering a crash when using UXCam SDK version 3.6.16 in our iOS app. The crash happens intermittently, and it appears to be related to the SDK's internal processes.

## Environment
- UXCam SDK Version: 3.6.16
- iOS Version: iOS 17.6.1
- Device: iPhone 13 and iPhone 12 Pro Max

## Crash Logs
```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x83f20 __exceptionPreprocess
1  libobjc.A.dylib                0x172b8 objc_exception_throw
2  CoreFoundation                 0x15e6c -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]
3  CoreFoundation                 0x15a88 +[NSDictionary dictionaryWithObjects:forKeys:count:]
4  Ordering App                   0xc5ba14 -[UXPlistHandler categorizeWaitingSessions]
5  Ordering App                   0xc5ac74 -[UXPlistHandler uploadAllWaitingSessions]
6  Ordering App                   0xc56720 __43-[UXPlistHandler beginUploadOfAllSessions:]_block_invoke
7  libdispatch.dylib              0x213c _dispatch_call_block_and_release
8  libdispatch.dylib              0x3dd4 _dispatch_client_callout
9  libdispatch.dylib              0xb400 _dispatch_lane_serial_drain
10 libdispatch.dylib              0xbf30 _dispatch_lane_invoke
11 libdispatch.dylib              0x16cb4 _dispatch_root_queue_drain_deferred_wlh
12 libdispatch.dylib              0x16528 _dispatch_workloop_worker_thread
13 libsystem_pthread.dylib        0x4934 _pthread_wqthread
14 libsystem_pthread.dylib        0x10cc start_wqthread
```

Please let us know if you need more information or have any suggestions to resolve this issue.

**Comment by gabrieluxcam** [comment]
Thank you for reporting this @srbalan  our team will review this asap and should there be any details required we will let you know. 

Thanks again and apologies for the hassle

**Comment by gabrieluxcam** [comment]
Hello @srbalan thanks for your patience in this matter. Our team has now fixed the crash and are finalising testing, so we can expect a release of the updated SDK around this Friday, I will of course let you know again as soon as it releases. 

Thanks!

**Comment by gabrieluxcam** [comment]
Hello @srbalan we have now released SDK version 3.6.17 with this fix, so it will be extremely helpful to upgrade and confirm if everything works as expected now. 

Thanks and apologies for the hassle!
